fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingThrottle.java

```diff
@@ -247,6 +265,15 @@ void refreshConfigLimits() {
                 recordsLimit = maxLimit;
             }
         }
+
+        private void addStoreTimerAtFailure(KeyValueLogMessage message) {
+            final FDBStoreTimer timer = common.getRunner().getTimer();
+            if (timer != null) {
+                StoreTimer metricsDiff = lastFailureSnapshot == null ? timer : StoreTimer.getDifference(timer, lastFailureSnapshot);
+                lastFailureSnapshot = StoreTimerSnapshot.from(timer);
```

Comment:
I don't think this will get the snapshot you want. Namely, this gathers the snapshot since the last failure, so if you have 1000 transactions successfully, and then fail, it will have the store timer for all 1001 transactions.
You could gather the snapshot at the beginning of every transaction, but that might be expensive. It might make sense to clear the snapshot after a success, and if you don't have one, gather it.

This way, if you have two failures in a row, it will include the snapshot, but if you have sporadic failures, that succeed on retry we won't log any snapshots.

Reply:
IIUIC, by clearing the snapshot after success the next displayed counters will be since the beginning of time (well, the indexing session). I had assumed that it will be better to have it this way - when you can see how many "things" had happened between failures and have a precise measurement for two consecutive ones.